acquisition.py

```diff
@@ -96,3 +108,144 @@ def get_reward_batch(self, inputs_af_base):  # inputs_af = list of ...
                 outputs = self.proxy.model(inputs, None)
 
         return outputs
+
+
+class AcquisitionFunctionUCB(AcquisitionFunctionBase):
+    def __init__(self, config, proxy):
+        super().__init__(config, proxy)
+
+    def load_best_proxy(self):
+        super().load_best_proxy()
+
+    def get_reward_batch(self, inputs_af_base):  # inputs_af = list of ...
+        super().get_reward_batch(inputs_af_base)
+
+        inputs_af = list(map(torch.tensor, inputs_af_base))
+        input_afs_lens = list(map(len, inputs_af_base))
+        inputs = pad_sequence(inputs_af, batch_first=True, padding_value=0.0)
+
+        self.load_best_proxy()
+        self.proxy.model.eval()
+        with torch.no_grad():
+            if self.config.proxy.model.lower() == "mlp":
+                outputs = (
+                    torch.hstack(
+                        [
+                            self.proxy.model(inputs)
+                            for _ in range(self.config.proxy.dropout_samples)
+                        ]
+                    )
+                    .cpu()
+                    .detach()
+                    .numpy()
+                )
+            elif self.config.proxy.model.lower() == "lstm":
+                outputs = (
+                    torch.hstack(
+                        [
+                            self.proxy.model(inputs, input_afs_lens)
+                            for _ in range(self.config.proxy.dropout_samples)
+                        ]
+                    )
+                    .cpu()
+                    .detach()
+                    .numpy()
+                )
+            elif self.config.proxy.model.lower() == "transformer":
+                outputs = (
+                    torch.hstack(
+                        [
+                            self.proxy.model(inputs, None)
```

Comment:
Do you think there would be any way to call the proxy in a very modular way ? like a general self.proxy.model(inputs, [optional parameters]) ? So that we don't have to distinguish in the acquisition function.

Reply:
Yeah I've already created a PR for that :)
Can only modify these files once that is merged